interfaces/apparmor/apparmor.go

```diff
@@ -83,22 +85,49 @@ func loadProfiles(fnames []string, cacheDir string, flags aaParserFlags) error {
 	return nil
 }
 
-// UnloadProfiles removes the named profiles from the running kernel.
-//
-// The operation is done with: apparmor_parser --remove $names...
-// The binary cache file is removed from /var/cache/apparmor
+// UnloadProfiles is meant to remove the named profiles from the running
+// kernel and then remove any cache files. Importantly, we can only unload
+// profiles when we are sure there are no lingering processes from the snap
+// (ie, forcibly stop all running processes from the snap). Otherwise, any
+// running processes will become unconfined. Since we don't have this guarantee
+// yet, leave the profiles loaded in the kernel but remove the cache files from
+// the system so the policy is gone on the next reboot.
 func unloadProfiles(names []string, cacheDir string) error {
 	if len(names) == 0 {
 		return nil
 	}
 
-	args := []string{"--remove"}
-	args = append(args, names...)
-	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("cannot unload apparmor profile: %s\napparmor_parser output:\n%s", err, string(output))
+	/* TODO: uncomment when no lingering snap processes is guaranteed
```

Comment:
sounds like this should have a matching bug in LP, no?

Reply:
Done